packages/elements/src/calendar/index.ts

```diff
@@ -1405,7 +1405,8 @@ export class Calendar extends ControlElement implements MultiValue {
       cancelable: false,
       composed: true, // allow calendar customization within other elements e.g. datetime picker
       detail: {
-        cell: calendarCell
+        cell: calendarCell,
+        calendar: this
```

Comment:
Do we need `calendar` in `before-cell-render` event?

Reply:
Yes, we do need to attach the `calendar` to know which calendar fires the event. Moreover, the developer can save the `calendar` as a reference to use in complex use cases.